kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -448,50 +444,45 @@ class ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier
         BaseType::mScaleFactor = this->GetScaleNorm(rModelPart, rA);
 
         // Detect if there is a line of all zeros and set the diagonal to a 1 if this happens
-        #pragma omp parallel firstprivate(system_size)
-        {
+        IndexPartition<std::size_t>(system_size).for_each([&](std::size_t Index){
             std::size_t col_begin = 0, col_end  = 0;
             bool empty = true;
 
-            #pragma omp for
-            for (int k = 0; k < static_cast<int>(system_size); ++k) {
-                col_begin = Arow_indices[k];
-                col_end = Arow_indices[k + 1];
-                empty = true;
-                for (std::size_t j = col_begin; j < col_end; ++j) {
-                    if(Avalues[j] != 0.0) {
-                        empty = false;
-                        break;
-                    }
+            col_begin = Arow_indices[Index];
+            col_end = Arow_indices[Index + 1];
+            empty = true;
+            for (std::size_t j = col_begin; j < col_end; ++j) {
+                if(Avalues[j] != 0.0) {
+                    empty = false;
+                    break;
                 }
+            }
 
-                if(empty) {
-                    rA(k, k) = BaseType::mScaleFactor;
-                    rb[k] = 0.0;
-                }
+            if(empty) {
+                rA(Index, Index) = BaseType::mScaleFactor;
```

Comment:
@philbucher 
I guess error is here, Is it not allowed ??? 
And, error is only for CentOS, Why is that a case, I am not able to understand it :(

Reply:
this should not happen, and it is definitely not threadsafe (it implies reshaping the CSR matrix).

@loumalouomega can u comment?

Reply:
@ashishdarekar I guess you need to capture it

Reply:
separately ?? `[&]` this includes everything right ?

Reply:
Solved. 
Thanks :)